### PR TITLE
Fix #478 Vagrantfile update for proxy support

### LINKED ISSUE #478: Vagrantfiles need update to accomodate proxy variable.
We need to update different vagrantfiles to make use of proxy variables.

### PR DESCRIPTION
**Pull request by praveenkumar**
This patch will add variables to support proxy and pass it to sccli.

### CODE REVIEW

File: components/centos/centos-openshift-setup/Vagrantfile [hunk 1]
```diff
@@ -33,6 +39,11 @@ end
 # For More info about Openshift, please refer to `offical documents
 # <https://docs.openshift.org/latest/welcome/index.html>`_.
 
+# Set default values if variable is not defined
+PROXY ||= ''
+PROXY_USER ||= ''
+PROXY_PASSWORD ||= ''
```
Comment (brgnepal) [inline components/centos/centos-openshift-setup/Vagrantfile:45]: :+1:

File: components/centos/centos-openshift-setup/Vagrantfile [hunk 2]
```diff
@@ -13,6 +13,12 @@ IMAGE_TAG="v1.2.0"
 # You will use this IP address to connect to OpenShift web console.
 PUBLIC_ADDRESS="10.1.2.2"
 
+# Http proxy setup
```
Comment (brgnepal) [inline components/centos/centos-openshift-setup/Vagrantfile]: HTTP ?

File: components/centos/centos-openshift-setup/Vagrantfile [hunk 3]
```diff
@@ -13,6 +13,12 @@ IMAGE_TAG="v1.2.0"
...
+  
+#PROXY="<proxy_server>:<port>"
```
Comment (brgnepal) [inline components/centos/centos-openshift-setup/Vagrantfile]: Space after `#`?

File: components/centos/centos-openshift-setup/Vagrantfile [hunk 4]
```diff
@@ -13,6 +13,12 @@ IMAGE_TAG="v1.2.0"
...
+#PROXY_USER="foo"
+#PROXY_PASSWORD="mysecertpass"
```
Comment (brgnepal) [inline components/centos/centos-openshift-setup/Vagrantfile]: typo in secret

File: components/centos/centos-openshift-setup/Vagrantfile (hunk 3 above)
Comment (brgnepal) [inline components/centos/centos-openshift-setup/Vagrantfile]: adding spaces around `=` will increase readability of sentence.

Also, please add space in DOCKER_REGISTRY, IMAGE_NAME and where ever you find such issue.

File: components/centos/centos-openshift-setup/Vagrantfile [hunk 5]
```diff
@@ -33,6 +39,11 @@ end
 # For More info about Openshift, please refer to `offical documents
 # <https://docs.openshift.org/latest/welcome/index.html>`_.
 
+# Set default values if variable is not defined
```
Comment (brgnepal) [inline components/centos/centos-openshift-setup/Vagrantfile]: Set default values for proxy settings if not defined

**Comment by LalatenduMohanty** [comment]
@praveenkumar can you please change the OSE Vagrantfile for CDK?

File: components/centos/centos-openshift-setup/Vagrantfile [hunk 6]
```diff
@@ -13,6 +13,12 @@ IMAGE_TAG="v1.2.0"
 # You will use this IP address to connect to OpenShift web console.
 PUBLIC_ADDRESS="10.1.2.2"
 
+# Variables to enable HTTP/HTTPS proxy
+  
+#PROXY = "<proxy_server>:<port>"
```
Comment (LalatenduMohanty) [inline components/centos/centos-openshift-setup/Vagrantfile:18]: We need to clearly mention how we expect the user to put proxy server information i.e. http://<ip>:<port> Vs <ip> :<port>. May be we can add a dummy example.

**Comment by LalatenduMohanty** [comment]
Tested and LGTM.

**Pull request merged**